docs/core/integration-quality-scale/rules/iqs012.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "IQS012 - Entities use has_entity_name = True"
+---
+
+IQS012 is as follows:
+
+Entities use has_entity_name = True
+
+## Reasoning
+
+`has_entity_name` is an entity attribute, which enables a new naming method.
+It is introduced to split the name of the entity from the name of the device or service.
+With `has_entity_name` set to `True`, the user will have a better experience on the device page, since not every entity is prefixed with the device name.
+This feature also allowed users to rename a device and automatically update all entities that belong to that device.
+
+## Example implementation
+
+In the example below, if the name of the device is "My device" and the field is "temperature", the name of the entity will be shown as "My device temperature".
+
+`sensor.py`
+```python
+class MySensor(SensorEntity):
+    """Representation of a sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, device: Device, field: str) -> None:
+        """Initialize the sensor."""
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.id)},
+            name=device.name,
+        )
+        self._attr_name = field
+```
+
+However, when the name of the entity is set to `None`, the name of the device will be used as the name of the entity.
```

Comment:
I'm conflicted about these examples and docs. It's good that you can understand the rule from reading a single page, but at the same time we're duplicating docs in multiple places which will make it harder to maintain.

Reply:
I agree with both points. But I also think that the other docs are written by devs and not for (new) devs, since all the information you need is there, but as a new dev you are missing too much context on what something is.

On one hand I am thinking, this is maybe good for the time being, and we should identify the points where we explain too much and maybe just move examples or extra practical context to the real documentation page